tuf/store/httpstore.go

```diff
@@ -270,13 +275,54 @@ func (s HTTPStore) buildURL(uri string) (*url.URL, error) {
 	return s.baseURL.ResolveReference(sub), nil
 }
 
-// GetKey retrieves a public key from the remote server
-func (s HTTPStore) GetKey(role string) ([]byte, error) {
+// GetKey retrieves the most recently created (whether it is signed in yet or not)
+// public key for the given role from the remote server
+func (s HTTPStore) GetKey(role string) (data.PublicKey, error) {
+	return s.requestKey(role, nil)
+}
+
+// RotateKey rotates a key on the remote server and returns the new public key.  This requires
+// a request with a short expiry time (to limit replay), signed by at least one root key.
+// Signing with the root key proves that the client making the request has the capability of
+// actually rotating the key, and is not just making a spurious rotation request.
+func (s HTTPStore) RotateKey(role string, cs signed.CryptoService, roots ...data.PublicKey) (data.PublicKey, error) {
+	req := &data.SignedCommon{
+		Type:    role,
+		Expires: time.Now().Add(5 * time.Minute),
+	}
+	reqJSON, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+	signedReq := &data.Signed{Signed: reqJSON}
+	if err := signed.Sign(cs, signedReq, roots...); err != nil {
```

Comment:
yeah... this feels like completely the wrong place to do this. The stores were very cohesive before. Now suddenly they're responsible for signing and json marshalling? This is how we end up with confusing code where it's not clear what happens where.